tools.py

```
@@ -107,75 +102,72 @@ def sign(templatePath, privateKeyPath):
         )
 
 
-# Function that checks signature againsed stringified filtered snippet
-def executeValidation(signedSnippet, encodedSignature, publicKeyPath):
+def executeValidation(signedSnippet, encodedSignature):
+    """
+    Function that checks signature againsed stringified filtered snippet
+    """
     serializedSnippet = pickle.dumps(signedSnippet)
     decodedSignature = base64.b64decode(encodedSignature)
 
-    # load public key
-    with open(publicKeyPath, 'rb') as public_key_file:
-        public_key = serialization.load_pem_public_key(
-            public_key_file.read(),
-        )
+    fd, fn = tempfile.mkstemp()
+    os.write(fd, decodedSignature)
+    os.close(fd)
 
-    # Validate Signature
-    result = public_key.verify(
-        decodedSignature,
-        serializedSnippet,
-        padding.PSS(
-            mgf=padding.MGF1(hashes.SHA256()),
-            salt_length=padding.PSS.MAX_LENGTH
-        ),
-        hashes.SHA256()
-    )
+    result = gpg.verify_data(fn, serializedSnippet)
```

I'm not sure if gpg throws exceptions, but if it does then you could leak this temp file.

---

Heya Derek!  So as far as exceptions are concerned, this module "python-gnupg" generally throws silent errors unless configured to log its data with the python "logging" module.  
https://gnupg.readthedocs.io/en/latest/#logging